crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -844,9 +844,9 @@ impl<S> StateMachineApplyContext<'_, S> {
 
         let record_unique_ts = UniqueTimestamp::from_unix_millis(self.record_created_at).unwrap();
         let visible_at = if let Some(execution_time) = metadata.execution_time {
-            VisibleAt::from_unix_millis(execution_time)
+            VisibleAt::new(execution_time)
         } else {
-            VisibleAt::At(record_unique_ts)
+            VisibleAt::new(self.record_created_at)
         };
```

Comment:
Maybe could be simplified to `let visible_at = VisibleAt::new(metadata.execution_time.unwrap_or(self.record_created_at))`.